lib/Doctrine/CouchDB/CouchDBClient.php

```diff
@@ -145,13 +146,38 @@ public function getUuids($count = 1)
     /**
      * Find a document by ID and return the HTTP response.
      *
-     * @param  string $id
+     * If $allOpenRevs is true, documents for all leaf revisions are
+     * retrieved and $revisions parameter is ignored. If $allOpenRevs
+     * is false and $revisions is not null, only the specified leaf revisions
+     * are retrieved.
+     *
+     * @param string $id
+     * @param bool $allOpenRevs
+     * @param array $revisions
      * @return HTTP\Response
      */
-    public function findDocument($id)
-    {
-        $documentPath = '/' . $this->databaseName . '/' . urlencode($id);
-        return $this->httpClient->request( 'GET', $documentPath );
+    public function findDocument(
```

Comment:
Using `allOpenRevs` or `revisions` changes the response format of the call or? In this case, we need a new method instead.